### PR TITLE
Apm stuff

### PR DESCRIPTION
**Pull request by tcanabrava**
Some Qt and C++ changes to modernize the code, feel free to take a look and discard if not desired.

### CODE REVIEW

File: src/AutoPilotPlugins/APM/APMSensorsComponent.cc [hunk 1]
```diff
@@ -117,7 +117,8 @@ bool APMSensorsComponent::compassSetupNeeded(void) const
     for (size_t i=0; i<cCompass; i++) {
         if (_autopilot->getParameterFact(FactSystem::defaultComponentId, devicesIds[i])->rawValue().toInt() != 0) {
             for (size_t j=0; j<cOffset; j++) {
-                if (_autopilot->getParameterFact(FactSystem::defaultComponentId, rgOffsets[i][j])->rawValue().toFloat() == 0.0f) {
```
Comment (DonLakeFlyer) [inline src/AutoPilotPlugins/APM/APMSensorsComponent.cc]: 0.0 is set directly as a default for the parameter by the firmware. We are checking to see if the value has been changed by the user to anything other than the default. Hence comparison to exact 0.0 is what we want here.

File: src/AutoPilotPlugins/APM/APMSensorsComponent.cc [hunk 2]
```diff
@@ -142,7 +143,8 @@ bool APMSensorsComponent::accelSetupNeeded(void) const
     for (size_t i=0; i<cAccel; i++) {
         if (_autopilot->getParameterFact(FactSystem::defaultComponentId, insUse[i])->rawValue().toInt() != 0) {
             for (size_t j=0; j<cOffset; j++) {
-                if (_autopilot->getParameterFact(FactSystem::defaultComponentId, rgOffsets[i][j])->rawValue().toFloat() == 0.0f) {
```
Comment (DonLakeFlyer) [inline src/AutoPilotPlugins/APM/APMSensorsComponent.cc]: Same thing here.

File: src/AutoPilotPlugins/PX4/PX4AutoPilotPlugin.cc [hunk 3]
```diff
@@ -94,32 +94,32 @@ const QVariantList& PX4AutoPilotPlugin::vehicleComponents(void)
         Q_ASSERT(_vehicle);
         
         if (parametersReady()) {
-            _airframeComponent = new AirframeComponent(_vehicle, this);
```
Comment (DonLakeFlyer) [inline src/AutoPilotPlugins/PX4/PX4AutoPilotPlugin.cc]: I'd change this to match the pattern you used in APMAutoPilotPlugin

**Comment by DonLakeFlyer** [comment]
@tcanabrava Needs a rebase and merge resolution.

**Comment by DonLakeFlyer** [comment]
@tcanabrava This pull is getting kind of stale

**Comment by tcanabrava** [comment]
I forgot about it. De-stalling it today.
Em 17 de jan de 2016 23:44, "Don Gagne" <notifications@github.com> escreveu:

**Pull request closed without merging**